src/main/java/com/softserve/app/controller/ForgotPasswordController.java

```diff
@@ -0,0 +1,101 @@
+package com.softserve.app.controller;
+
+import com.softserve.app.models.User;
+import com.softserve.app.request.EmailRequest;
+import com.softserve.app.request.ResetPasswordRequest;
+import com.softserve.app.service.userService.UserServiceImpl;
+import net.bytebuddy.utility.RandomString;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.repository.query.Param;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import javax.servlet.http.HttpServletRequest;
+import java.io.UnsupportedEncodingException;
+
+//@CrossOrigin(origins = "http://localhost:3000")
+@RestController
+public class ForgotPasswordController {
+    @Autowired
+    private JavaMailSender mailSender;
+
+    @Autowired
+    private UserServiceImpl userService;
+
+    @PostMapping("/forgot_password")
+    public ResponseEntity<?> processForgotPassword(@RequestBody EmailRequest emailRequest, HttpServletRequest request, Model model) {
+        String email = emailRequest.getEmail();
+        String token = RandomString.make(30);
+        try {
+            userService.updateResetPasswordToken(token, email);
+            String resetPasswordLink = "http://localhost:3000/reset_password?token=" + token;
+            sendEmail(email, resetPasswordLink);
+        } catch (UnsupportedEncodingException | MessagingException e) {
+
+            return new ResponseEntity<>("Failed to send an email", HttpStatus.BAD_REQUEST);
```

Comment:
throw custom your exception (controller advice will handle this exception and create an entity with HttpStatus.